src/path/mod.rs

```diff
@@ -348,13 +398,22 @@ where
     I: Into<PathPart<'a>>,
 {
     fn from_iter<T: IntoIterator<Item = I>>(iter: T) -> Self {
-        let raw = T::into_iter(iter)
-            .map(|s| s.into())
-            .filter(|s| !s.raw.is_empty())
-            .map(|s| s.raw)
-            .join(DELIMITER);
+        let mut this = Self::ROOT;
+        this.extend(iter);
+        this
+    }
+}
 
-        Self { raw }
+impl<'a, I: Into<PathPart<'a>>> Extend<I> for Path {
+    fn extend<T: IntoIterator<Item = I>>(&mut self, iter: T) {
+        for s in iter.into_iter() {
+            let s = s.into();
+            if s.raw.is_empty() {
+                continue;
+            }
+            self.raw.push(DELIMITER_CHAR);
```

Comment:
Can you add some tests showing how this works for:
1. Empty iterators
2. Iterator with a single element 
3. Iterator with multiple elements
4. Extending an existing (non empty) Path? (it looks like maybe this will add an extra `/` 🤔 )